Realm/RLMRealm.mm

```diff
@@ -780,4 +786,165 @@ - (void)detachAllEnumerators {
     _collectionEnumerators = nil;
 }
 
+#pragma mark - Async Writes
+
+- (void)writeAsyncWithBlock:(RLMAsyncWriteBlock)block completion:(RLMCompletionBlock)completion {
+    RLMRealmConfiguration *config = self.configuration;
+    
+    __block NSError *error;
+    __block BOOL didComplete;
+    
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+    __block RLMNotificationToken *token = [self addNotificationBlock:^(NSString *notification, RLMRealm *realm) {
```

Comment:
Without __block it captures the value of the variable at the time the block is created (i.e. nil, since the block is created before addNotificationBlock is called).